controllers/component_image_controller.go

```diff
@@ -108,7 +113,12 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, fmt.Errorf("error reading component: %w", err)
 	}
 
+	componentIdForMetrics := fmt.Sprintf("%s=%s", component.Name, component.Namespace)
```

Comment:
I'd create a separate function `getComponentIdForMetrics(component)`. This way, we won't need to change signature of `reportError` method.

Reply:
done